src/Analysis/Ast/Impl/Specializations/Typing/Values/TypingType.cs

```diff
@@ -28,10 +28,12 @@ internal sealed class TypingType : LocatedMember, IPythonType {
         public TypingType(IPythonModule declaringModule, IPythonType type): base(declaringModule) {
             _type = type ?? throw new ArgumentNullException(nameof(type));
             Name = $"Type[{_type.Name}]";
+            QualifiedName = $"{DeclaringModule.Name}:Type[{_type.QualifiedName}]";
```

Comment:
Can `DeclaringModule` be null or have no name?

Reply:
No, should never be null. And modules always have names.

Reply:
Can we get here from unsaved file?

Reply:
Yes but it is specialized typing so it is always actually `typing`.